src/Compilers/Core/Portable/Operations/ControlFlowGraphBuilder.cs

```diff
@@ -3679,6 +3679,66 @@ public override IOperation VisitInstanceReference(IInstanceReferenceOperation op
             }
         }
 
+        public override IOperation VisitDynamicInvocation(IDynamicInvocationOperation operation, int? captureIdForResult)
+        {
+            (IOperation rewrittenOperation, ImmutableArray<IOperation> rewrittenArguments) = VisitDynamicOperationWithArguments(operation.Operation, operation.Arguments);
+            return new DynamicInvocationExpression(rewrittenOperation, rewrittenArguments, ((HasDynamicArgumentsExpression)operation).ArgumentNames,
+                ((HasDynamicArgumentsExpression)operation).ArgumentRefKinds, semanticModel: null, operation.Syntax, operation.Type, operation.ConstantValue, operation.IsImplicit);
+        }
+
+        public override IOperation VisitDynamicIndexerAccess(IDynamicIndexerAccessOperation operation, int? captureIdForResult)
+        {
+            (IOperation rewrittenOperation, ImmutableArray<IOperation> rewrittenArguments) = VisitDynamicOperationWithArguments(operation.Operation, operation.Arguments);
```

Comment:
>VisitDynamicOperationWithArguments [](start = 93, length = 34)

It doesn't look like we can use this helper for IDynamicIndexerAccessOperation, it looks like ```Operation``` is considered a value for the purpose of indexing and it should be spilled in case of a control flow in the arguments

Reply:
Fixed, and verified with test baseline change for `DynamicIndexerAccess_ControlFlowWithDynamicReceiver`

---
In reply to: [187246124](https://github.com/dotnet/roslyn/pull/26690#discussion_r187246124) [](ancestors = 187246124)